app/forms/DatePageFormProvider.scala

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package forms
+
+import forms.mappings.Mappings
+import play.api.data.Form
+
+import java.time.LocalDate
+import javax.inject.Inject
+
+class DatePageFormProvider @Inject() extends Mappings {
```

Comment:
I was thinking that maybe the provider could return a case class that wraps the form and the view can accept that as a parameter. This would mean that a dev couldn't create a form with the wrong fields but still satisfys the type signature. This will need some thought as form as methods that are needed and would break it were to be a parameter on another case class. Perhaps we can go through this

Reply:
Yeah it would be good to go through this

Reply:
Yes, we can go through this

Reply:
Had a chat and agreed to leave this for now